code/game/objects/items/sticker.dm

```diff
@@ -0,0 +1,226 @@
+/// parent type for all other stickers. do not spawn directly
+/obj/item/sticker
+	name = "sticker"
+	desc = "A sticker with some strong adhesive on the back, sticks to stuff!"
+	item_flags = NOBLUDGEON | XENOMORPH_HOLDABLE //funny
+	resistance_flags = FLAMMABLE
+	icon = 'icons/obj/stickers.dmi'
+	w_class = WEIGHT_CLASS_TINY
+	throw_range = 3
+	vis_flags = VIS_INHERIT_DIR | VIS_INHERIT_PLANE | VIS_INHERIT_LAYER
+	///The overlay we apply to things we stick to
+	var/mutable_appearance/sticker_overlay
+	///A list of icon_states to pick an icon_state on Initialize, provided it is not null.
+	var/list/icon_states
+	///The thing we are attached to
+	var/atom/attached
+	///The turf our COMSIG_TURF_EXPOSE is registered to, so we can unregister it later.
+	var/turf/signal_turf
+	/// If the sticker should be disincluded from normal sticker boxes.
+	var/contraband = FALSE
+
+/obj/item/sticker/Initialize(mapload)
+	. = ..()
+	if(icon_states)
+		icon_state = pick(icon_states)
+	pixel_y = rand(-3,3)
+	pixel_x = rand(-3,3)
+
+/obj/item/sticker/afterattack(atom/target, mob/living/user, prox, params)
+	. = ..()
+	if(!prox)
+		return
+	if(!isliving(target) && !isobj(target) && !isturf(target))
+		return
+	var/list/parameters = params2list(params)
+	if(!LAZYACCESS(parameters, ICON_X) || !LAZYACCESS(parameters, ICON_Y))
+		return
+	var/divided_size = world.icon_size / 2
+	var/px = text2num(LAZYACCESS(parameters, ICON_X)) - divided_size
+	var/py = text2num(LAZYACCESS(parameters, ICON_Y)) - divided_size
+	. |= AFTERATTACK_PROCESSED_ITEM
+	user.do_attack_animation(target)
+	stick(target,user,px,py)
+	return .
+
+///Sticks this sticker to the target, with the pixel offsets being px and py.
+/obj/item/sticker/proc/stick(atom/target, mob/living/user, px,py)
+	sticker_overlay = mutable_appearance(icon, icon_state , layer = target.layer + 1, appearance_flags = RESET_COLOR | PIXEL_SCALE)
+	sticker_overlay.pixel_x = px
+	sticker_overlay.pixel_y = py
+	target.add_overlay(sticker_overlay)
+	attached = target
+	if(isliving(target) && user)
+		var/mob/living/victim = target
+		if(victim.client)
+			user.log_message("stuck [src] to [key_name(victim)]", LOG_ATTACK)
+			victim.log_message("had [src] stuck to them by [key_name(user)]", LOG_ATTACK)
+	register_signals(user)
+	moveToNullspace()
```

Comment:
why not move the sticker into them?

Reply:
You won't be able to put a sticker on a closet. It's already going to look weird but this is going to shove it inside.